tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py

```diff
@@ -205,7 +333,141 @@ def test_plot_parallel_coordinate_unique_hyper_param() -> None:
         )
     )
 
-    # still "category_a" contains unique suggested value during the optimization.
+    # Still "category_a" contains unique suggested value during the optimization.
     figure = plot_parallel_coordinate(study_categorical_params)
-    assert len(figure.get_lines()) == 0
+    axes = figure.get_figure().axes
+    assert len(axes) == 3 + 1
+    assert axes[0].get_ylim() == (0.0, 2.0)
+    assert axes[1].get_ylabel() == "Objective Value"
+    assert axes[1].get_ylim() == (0.0, 2.0)
+    objectives = _fetch_objectives_from_figure(figure)
+    # Objective values are not sorted by the other parameters,
+    # unlike Plotly's parallel_coordinate.
+    assert objectives == [0.0, 2.0]
+    assert axes[2].get_ylim() == (-default_padding_fraction, default_padding_fraction)
+    assert [lien.get_text() for lien in axes[2].get_yticklabels()] == ["preferred"]
+    assert [label.get_position()[1] for label in axes[2].get_yticklabels()] == [0]
+    assert axes[3].get_ylim() == (20, 30)
+    expected_labels = ["Objective Value", "category_a", "param_b"]
+    _test_xtick_labels(axes, expected_labels)
+    plt.savefig(BytesIO())
+
+
+def test_plot_parallel_coordinate_with_categorical_numeric_params() -> None:
+    # Test with sample from multiple distributions including categorical params
+    # that can be interpreted as numeric params.
+    study = create_study()
+    study.add_trial(
+        create_trial(
+            value=0.0,
+            params={"param_a": "preferred", "param_b": 2, "param_c": 30, "param_d": 2},
+            distributions={
+                "param_a": CategoricalDistribution(("preferred", "opt")),
+                "param_b": CategoricalDistribution((1, 2, 10)),
+                "param_c": FloatDistribution(1, 1000),
+                "param_d": CategoricalDistribution((1, -1, 2)),
+            },
```

Comment:
How about pre-defining the distributions and use it in the following calls of `create_trial` to reduce the number of lines?

Reply:
Love it. This code comes from Plotly's test, so I've also updated Plotly's test too.